plugins/dialog/plugin.js

```diff
@@ -848,10 +843,13 @@ CKEDITOR.DIALOG_STATE_BUSY = 2;
 			// Insert the dialog's element to the root document.
 			var element = this._.element;
 			var definition = this.definition;
-			if ( !( element.getParent() && element.getParent().equals( CKEDITOR.document.getBody() ) ) )
+
+			if ( !( element.getParent() && element.getParent().equals( CKEDITOR.document.getBody() ) ) ) {
 				element.appendTo( CKEDITOR.document.getBody() );
-			else
-				element.setStyle( 'display', 'block' );
+			} else {
+				var display = ( !CKEDITOR.env.ie && CKEDITOR.env.edge ) ? 'block' : 'flex';
+				element.setStyle( 'display', display );
```

Comment:
You could just pass it directly into function execution e.g.
```javascript
element.setStyle( 'display', !CKEDITOR.env.ie && CKEDITOR.env.edge ? 'block' : 'flex' );
```

and additional condition brackets are redundant.